packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -0,0 +1,151 @@
+"""Utilizes command line args to create a Measurement Plug-In Client using template files."""
+
+import pathlib
+from typing import Any, Dict, Optional
+
+import black
+import click
+from mako.template import Template
+from ni_measurement_plugin_sdk_service._internal.stubs.ni.measurementlink.measurement.v2 import (
+    measurement_service_pb2 as v2_measurement_service_pb2,
+)
+from ni_measurement_plugin_sdk_service.discovery import DiscoveryClient
+from ni_measurement_plugin_sdk_service.grpc.channelpool import GrpcChannelPool
+
+from ni_measurement_plugin_sdk_generator.client._constants import (
+    V2_MEASUREMENT_SERVICE_INTERFACE,
+)
+from ni_measurement_plugin_sdk_generator.client._support import (
+    _get_configuration_metadata_by_index,
+    _get_configuration_parameters_with_type_and_default_values,
+    _get_measurement_service_stub,
+    _get_output_metadata_by_index,
+    _get_output_parameters_with_type,
+    _get_python_module_name,
+)
+
+
+_IMPORT_MODULES: Dict[str, str] = {}
+
+
+def _render_template(template_name: str, **template_args: Any) -> bytes:
+    file_path = str(pathlib.Path(__file__).parent / "templates" / template_name)
+    template = Template(filename=file_path, input_encoding="utf-8", output_encoding="utf-8")
+
+    try:
+        return template.render(**template_args)
+    except Exception:
+        raise RuntimeError(f'An error occurred while rendering template "{template_name}".')
+
+
+def _create_file(
+    template_name: str, file_name: str, directory_out: pathlib.Path, **template_args: Any
+) -> None:
+    output_file = directory_out / file_name
+
+    output = _render_template(template_name, **template_args).decode("utf-8")
+    formatted_output = black.format_str(
+        src_contents=output,
+        mode=black.Mode(line_length=100),
+    )
+
+    with output_file.open("wb") as file:
+        file.write(formatted_output.encode("utf-8"))
+
+
+@click.command()
+@click.argument("module_name", type=str, default="")
+@click.option(
+    "-s",
+    "--measurement-service-class",
+    help="The service class of your measurement.",
+)
+@click.option(
+    "-o",
+    "--directory-out",
+    help="Output directory for Measurement Plug-In Client files. Default: '<current_directory>/<module_name>'",
+)
+def create_client(
+    module_name: str,
+    measurement_service_class: str,
+    directory_out: Optional[str],
+) -> None:
+    """Generates a Python Measurement Plug-In Client module for the measurement service.
+
+    You can use the generated module to interact with the corresponding measurement service.
+
+    MODULE_NAME: Name for the Python Measurement Plug-In Client module to be generated.
```

Comment:
For the plugin generator, display_name is a positional argument because it's the only required command line option/argument. Everything else is optional.

The same is not true for module_name. It would make more sense for module_name to have a --option. And since service_class is required, maybe it should be a positional argument.

Reply:
Here, service_class is also an optional parameter. There is another feature ([--interactive mode](https://dev.azure.com/ni/DevCentral/_git/ASW?path=%2FSpecs%2FTest%20Automation%20Frameworks%2FE2719769_Support%20for%20Measurement%20Plug-In%20client%20APIs%2FE2719769_Support%20for%20Measurement%20Plug-In%20client%20APIs.md&version=GBmain&_a=preview&anchor=interactive-approach)) is planned to be implemented. 
Module name is common for both one-line approach and interactive mode , so, made this as a positional argument. But, we are not restricting users to give empty module_name, if your gives invalid or empty, service class name of the measurement is used to create module_name.

Reply:
Sure, I have made `module-name` as option. Also, I have added a validation for service class, to restrict None.

Reply:
ok